### ISSUE TITLE
Getting Started Guide should not lead to warnings

### ISSUE THREAD

**Issue by dseynhae** [intel/llvm#8843]
Following the directions in the [Getting Started with oneAPI](https://intel.github.io/llvm-docs/GetStartedGuide.html#getting-started-with-oneapi-dpc):

[BUILD DPC++ toolchain](https://intel.github.io/llvm-docs/GetStartedGuide.html#build-dpc-toolchain):

```bash
python $DPCPP_HOME/llvm/buildbot/configure.py
```

Configuring on SuSE12 gives me the following warning:

```txt
CMake Warning:
  Manually-specified variables were not used by the project:

    LIBCLC_GENERATE_REMANGLED_VARIANTS
    LIBCLC_TARGETS_TO_BUILD
```

This simply means that our Python scripts define those variables, but they are not used by the script.

But if they are not used, then why are they defined?

A good User Experience (UX) would be an implementation that does not generate warnings at all: don't define the variables if they are not relevant in the first place.

**To Reproduce**
Please describe the steps to reproduce the behavior:

1. Include code snippet as short as possible
2. Specify the command which should be used to compile the program
3. Specify the comment which should be used to launch the program
4. Indicate what is wrong and what was expected

**Environment (please complete the following information):**

- OS: Linux SuSE12
- Target device and vendor: NA
- DPC++ version: NA (GitHub sycl branch)
- Dependencies version: NA

**Additional context**

**Comment by steffenlarsen** [comment]
Should be addressed by https://github.com/intel/llvm/pull/13698.
